src/main/java/frc/robot/subsystems/Conveyor.java

```diff
@@ -25,9 +41,74 @@
   public Conveyor()
   {
     // instantiate hardware
+    timer = new Timer();
+    conveyorMotor = new PWMSparkMax(CONVEYOR_SPARK);
+    intakeSwitch = new DigitalInput(INTAKE_SWITCH);
+    conveyorSwitch = new DigitalInput(CONVEYOR_SWITCH);
+    shooterSwitch = new DigitalInput(SHOOTER_SWITCH);
+
   }
 
   // METHODS //
+  public boolean getIntakeSwitch() {
+    return intakeSwitch.get();
+  }
+
+  public boolean getConveyorSwitch() {
+    return conveyorSwitch.get();
+  }
+
+  public boolean getShooterSwitch() {
+    return shooterSwitch.get();
+  }
+
+  /*
+  standy() allows the conveyor belt to be on 
+  a standby mode. The conveyor will only spin 
+  unless a ball is given in it. If the bay is full, 
+  this function also contains code to prevent the 
+  loading of more new balls. 
+  */
+
+
+  public void standby(boolean shooterRunning) {
+    if(!shooterSwitch.get() && intakeSwitch.get()) {
+      feed(shooterRunning);
+
+      timer.reset();
+      timer.start();
+      while(!conveyorSwitch.get() && timer.get() < CONVEYOR_TIMEOUT);
```

Comment:
Is the conveyor supposed to be moving during this loop?

Reply:
Good question.  Basically the idea is to have the conveyor waiting for a ball to appear at the end of the intake.  Once it sees one (`intakeSwitch.get()`) then it will start spinning the conveyor (`feed()`) until the ball hits the upper switch (`conveyorSwitch`).  So basically the conveyor will auto queue up the balls as they come in through the intake.  So to answer your question, yea the conveyor would be spinning during this `while` loop and then stops once the ball passes the `conveyorSwitch`.

Reply:
Also the `timer` is there to prevent the robot from getting stuck in this while loop forever, in case the switches malfunction or something.  Cause if it gets stuck, the driver and operator lose control over the robot :(

Reply:
I think the `standby()` function is a little advanced for now, since we don't even have switches on the robot yet.  So I'd suggest to just get the basic conveyor stuff working (spin and reverse) and we can figure out this stuff later :)

Reply:
Always a good idea to have a backup timeout in case the next limit switch is not hit within a certain time. I'm not too sure I understand what is going on with these while loops in this `standby()` function, would try to rewrite it to be a bit more clear.

Reply:
Does this just hang at the while loop then until the timeout? Since wouldn't the standby function be called repeatedly during `execute()` in the command, just wondering if hanging it there would cause any issues. If it is a short time, should be ok. Also we had some issues with the limit switches accidentally triggering while the robot bounced around so hopefully, that doesn't happen again with you guys.